api/datasets.go

```diff
@@ -497,6 +410,141 @@ func (h *DatasetHandlers) initHandler(w http.ResponseWriter, r *http.Request) {
 	json.NewEncoder(w).Encode(env)
 }
 
+func formFileDataset(dsp *dataset.DatasetPod, r *http.Request) (cleanup func(), err error) {
+	var rmFiles []*os.File
+	cleanup = func() {
+		for _, f := range rmFiles {
+			// TODO - log error?
+			os.Remove(f.Name())
+		}
+	}
+
+	datafile, dataHeader, err := r.FormFile("file")
+	if err != nil && err != http.ErrMissingFile {
+		err = fmt.Errorf("error opening dataset file: %s", err)
+		return
+	}
+	if datafile != nil {
+		switch strings.ToLower(filepath.Ext(dataHeader.Filename)) {
+		case ".yaml", ".yml":
+			var data []byte
+			data, err = ioutil.ReadAll(datafile)
+			if err != nil {
+				err = fmt.Errorf("error reading dataset file: %s", err)
+				return
+			}
+			if err = dsutil.UnmarshalYAMLDatasetPod(data, dsp); err != nil {
+				err = fmt.Errorf("error unmarshaling yaml file: %s", err)
+				return
+			}
+		case ".json":
+			if err = json.NewDecoder(datafile).Decode(dsp); err != nil {
+				err = fmt.Errorf("error decoding json file: %s", err)
+				return
+			}
+		}
+	}
+
+	tfFile, _, err := r.FormFile("transform")
+	if err != nil && err != http.ErrMissingFile {
+		err = fmt.Errorf("error opening transform file: %s", err)
+		return
+	}
+	if tfFile != nil {
+		// TODO - this assumes a skylark / starlark transform file
+		f, e := ioutil.TempFile("", "transform")
+		if e != nil {
+			err = e
```

Comment:
Instead of needing this assignment, just call the error variable "err" 2 lines up.